pkg/cli/cmds/agent_service_windows.go

```diff
@@ -22,6 +22,14 @@ const (
 	defaultServiceDescription = "Rancher Kubernetes Engine v2 (agent) see docs https://github.com/rancher/rke2#readme"
 )
 
+var agentFlag = []cli.Flag{
+	&cli.StringFlag{
+		Name:   "cni",
+		Usage:  "(networking/windows) CNI Plugins to configure on windows only calico and none are supported (default: calico)",
+		EnvVar: "RKE2_CNI",
+	},
+}
+
```

Comment:
CNI selection is a server option. We should not add CNI configuration flags to agents. If the agent needs to know what the CNI is, it needs to either read it from server config, or detect it via the presence of specific cluster resources.